plugin/storage/es/spanstore/reader.go

```diff
@@ -201,12 +193,12 @@ func addRemoteReadClusters(fn timeRangeIndexFn, remoteReadClusters []string) tim
 	}
 }
 
-func getSourceFn(archive bool, maxDocCount int) sourceFn {
+func getSourceFn(useReadWriteAliases bool, maxDocCount int) sourceFn {
 	return func(query elastic.Query, nextTime uint64) *elastic.SearchSource {
 		s := elastic.NewSearchSource().
 			Query(query).
 			Size(maxDocCount)
-		if !archive {
+		if !useReadWriteAliases {
```

Comment:
this looks a bit suspicious. If you have a query that specifies the time range, e.g. `(t-1d, t)`, and the index rotation frequency is 1 day with 7 days retention, then the read alias will cover all 7 days. It's more efficient to give the time range to the query because it could (theoretically) perform filtering of the indices based on the time range first and not search all 7 of them.

So question: do we have to exclude these two clauses? What's the upstream call path to this function, does it include both FindTraces and GetTrace? Where does the GetTrace get the time frame (since we didn't have it as part of GetTrace API until very recently)?

Reply:
Was just reading about ES aliases and I may be wrong, it doesn't look like it actually optimizes which indices to search. But since each index will have an inverted index no the time, ES would be able to quickly discard those indices if the query itself contains a `date_range`. So my question still stands - should we be really dropping the `nextTime` filter?

Reply:
@yurishkuro [Here](https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/es/spanstore/reader.go#L238-L239) is where the `GetTrace` function gets the time frame. It subtracts the max span age from the current time to get the start time and then use the current time as the end time (and then looks in [1 hour past in each direction](https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/es/spanstore/reader.go#L346))

Reply:
I pushed up a change that keeps the filter to see how the CI reacts. I think I tried this before and the integration tests complained but that may have been due to a different reason.

Reply:
@yurishkuro It looks good!